### ISSUE TITLE
Improve error message returned on geo_search()

### ISSUE THREAD

**Issue by Kydlaw** [tweepy/tweepy#1427]
### Issue
When using the `.geo_search()` method, if the arguments provided are incorrect (in my case, I put the coordinates in a list like I do in the `.search()` method), the resulting error is unclear:

```
Traceback (most recent call last):
  File "c:/Users/Kydlaw/Code/twitter-mapper/twitter_mapper/extraction.py", line 21, in <module>
    results = api.geo_search("40.712728", "-74.006015")
  File "C:\Users\Kydlaw\AppData\Local\pypoetry\Cache\virtualenvs\twitter-mapper-EGq-388k-py3.8\lib\site-packages\tweepy\binder.py", line 252, in _call
    return method.execute()
  File "C:\Users\Kydlaw\AppData\Local\pypoetry\Cache\virtualenvs\twitter-mapper-EGq-388k-py3.8\lib\site-packages\tweepy\binder.py", line 238, in execute
    result = self.parser.parse(self, resp.text, return_cursors=self.return_cursors)
  File "C:\Users\Kydlaw\AppData\Local\pypoetry\Cache\virtualenvs\twitter-mapper-EGq-388k-py3.8\lib\site-packages\tweepy\parsers.py", line 98, in parse
    result = model.parse_list(method.api, json)
  File "C:\Users\Kydlaw\AppData\Local\pypoetry\Cache\virtualenvs\twitter-mapper-EGq-388k-py3.8\lib\site-packages\tweepy\models.py", line 486, in parse_list
    item_list = json_list['result']['places']
KeyError: 'result'
```
In my case, I provided the coordinates as a list of str instead of lat and long.
The resulting response is an error code 12.

### Suggestion:

- Label for this issue: Improvement/Documentation

Catch the error code 12 case and raise an exception with an explicit message.
Or return the error message provided into the json.

**Comment by jiyoungsin** [comment]
This change might change everyone else's working code. I am not sure how we can resolve this issue since it has been so long since it was built.

**Comment by jiyoungsin** [comment]
The only solution I could suggest is to wrap your ". geo_search()" call in a try catch block and catch it on your side.

**Comment by Harmon758** [comment]
This should now be resolved with https://github.com/tweepy/tweepy/commit/c7847bac01e4f5c67f7960d0eea85f4e32130315.

> This change might change everyone else's working code.

A `KeyError` being raised here is unexpected and unsupported behavior. Being dependent on this is akin to [spacebar heating](https://xkcd.com/1172/).
If you were previously handling it, you can simply handle the `TweepyException` that's raised now.
Otherwise, there shouldn't be any other differences.
Regardless, this is set to be released with v4.0 with other much larger breaking changes anyway.
